chain/ethereum/src/adapter.rs

```diff
@@ -500,6 +602,19 @@ pub(crate) struct EthereumBlockFilter {
     pub trigger_every_block: bool,
 }
 
+impl Into<Vec<CallToFilter>> for EthereumBlockFilter {
+    fn into(self) -> Vec<CallToFilter> {
+        self.contract_addresses
+            .iter()
+            .dedup_by(|x, y| x.1 == y.1)
```

Comment:
`dedup_by` only dedups consecutive elemements, so it doesn't ensure no duplicates.

Reply:
sorted